presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrder.java

```diff
@@ -284,36 +283,88 @@ private void checkEncodeInputValidity(List<Integer> input, int maximumBits)
         checkArgument((totalBitLength <= maximumBits), format("The z-address type specified is not large enough to hold %d bits.", totalBitLength));
     }
 
+    /**
+     * The BoundedZValueRange class contains non-null Integer minimum and maximum value lists. The bounds are all inclusive.
+     */
+    private static class BoundedZValueRange
```

Comment:
Sounds like BoundedZValueRange and ZValueRange mean similarly. What is the reason that we need BoundedZValueRange?

Reply:
`BoundedZValueRange` uses an unboxed int type, so it rules out the possibility of null values.
`ZValueRange` can have null values for the intervals, but they should become bounded after filling in nulls because we only want to operate on the bounded ranges.